src/util/arch/x86/simd_utils.h

```diff
@@ -178,13 +180,11 @@ m128 load_m128_from_u64a(const u64a *p) {
 
 #define CASE_RSHIFT_VECTOR(a, count)  case count: return _mm_srli_si128((m128)(a), (count)); break;
 
+// we encounter cases where an argument slips past __builtin_constant_p but 
+// still fails to meet the (stricter) criteria demanded by the underlying 
+// intrinsic. in those cases we want to explicitly avoid the optimization.
 static really_inline
-m128 rshiftbyte_m128(const m128 a, int count_immed) {
-#if defined(HAVE__BUILTIN_CONSTANT_P) && !defined(VS_SIMDE_BACKEND)
-    if (__builtin_constant_p(count_immed)) {
-        return _mm_srli_si128(a, count_immed);
-    }
-#endif
+m128 rshiftbyte_m128_notconst(const m128 a, int count_immed) {
```

Comment:
this should be renamed rshiftbyte_m128_nim (not immediate), it's not a const, it's an immediate value that goes straight into the assembly instruction opcode in the relevant bits